File: src/python/classes/model_parameters.py
import dataclasses
from dataclasses import dataclass
from typing import Dict

@dataclass
class StochasticParameters:
    """
    Parameters for the stochastic processes used in the tolling agreement valuation.
    """
    
    sigma_g: float
    """Volatility of gas price (annualized, e.g., 0.4 for 40%). Calibrated to GBM."""
    
    sigma_p: float
    """Volatility of power price (annualized). Diffusion component of MRJD."""
    
    kappa: float
    """Mean reversion speed for power prices (unitless). High value means fast reversion."""
    
    lambda_j: float
    """Jump intensity for power prices (expected number of jumps per year)."""
    
    mu_j: float
    """Mean of the log-jump size for power prices."""
    
    sigma_j: float
    """Standard deviation of the log-jump size for power prices."""
    
    rho: float
    """Correlation between the Brownian motions of Gas and Power."""
    
    r: float
    """Risk-free interest rate (annualized, e.g., 0.02 for 2%)."""

    def to_dict(self) -> Dict:
        """Converts the parameters to a dictionary."""
        return dataclasses.asdict(self)
    
    @staticmethod
    def from_dict(dictionary: Dict) -> "StochasticParameters":
        """Creates a StochasticParameters instance from a dictionary."""
        return StochasticParameters(**dictionary)

File: src/rust/tolling_agreement_valuation/src/core/parameters/model_parameters.rs
use serde::Deserialize;

#[derive(Debug, Deserialize, Clone)]
pub struct ModelParameters<T> {
    // Volatility of gas price (in percent)
    pub sigma_g: T,
    // Volatility of power price (in percent)
    pub sigma_p: T,
    // mean reversion speed (unitless)
    pub kappa: T,
    // number of jumps per day (jumps / day)
    pub lambda_j: T,
    // mean jump height of power price (€ / mwh)
    pub mu_j: T,
    // variance of jump height (€ / mwh)
    pub sigma_j: T,
    // correlation of gas- and power prices (€ / mwh^2)
    pub rho: T,
}

impl<T> ModelParameters<T> {
    #[allow(clippy::too_many_arguments)]
    pub fn new(
        sigma_g: T,
        sigma_p: T,
        kappa: T,
        lambda_j: T,
        mu_j: T,
        sigma_j: T,
        rho: T,
    ) -> Self {
        ModelParameters {
            sigma_g,
            sigma_p,
            kappa,
            lambda_j,
            mu_j,
            sigma_j,
            rho,
        }
    }
}
